apps/src/code-studio/pd/application/teacher1920/Section3TeachingBackground.jsx

```diff
@@ -36,8 +36,13 @@ export default class Section3TeachingBackground extends LabeledFormComponent {
     'currentRole'
   ];
 
-  handleSchoolChange = selectedSchool =>
-    this.handleChange({school: selectedSchool && selectedSchool.value});
+  handleSchoolChange = selectedSchool => {
+    this.handleChange({
+      school: selectedSchool && selectedSchool.value,
+      schoolZipCode:
```

Comment:
instead of storing this separately, can't we get the zip code off of the school when we need it? `this.props.data.school && this.props.data.school.zip`

Reply:
Let me check. I thought `[Teacher1920Application].state.data.school` is just the school id and is different from `[Teacher1920Application].state.data.schoolZipCode` which comes from custom field.

Reply:
Oh yeah you might be right about that. If that’s the case then this looks good!

Reply:
Oh yeah you might be right about that. If that’s the case then this looks good!

Reply:
Okay, so the flow looks like this
1. In SchoolAutocompleteDropdown component, `getOptions` function sends request `GET /dashboardapi/v1/schoolsearch/:q/:limit` to schools_controller.rb, which calls `Api::V1::SchoolAutocomplete.get_matches`, which then returns a JSON with `attributes :nces_id, :name, :city, :state, :zip, :school_type, :latitude, :longitude`

2. The returned result then be re-packed by `constructSchoolOption` function. Its set `value` to school's nces_id, `label` to a combination of school's name, city, state, and zip, and keep the original JSON return in `school`.

3. Section3TeachingBackground's `handleSchoolChange` is triggered (which is this change), and unpack value in step 2 into school Id and school zip code.